staging/src/k8s.io/apimachinery/pkg/util/waitgroup/optionalwaitgroup.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package waitgroup
+
+import (
+	"sync"
+)
+
+// OptionalWaitGroup is a WaitGroup or nothing.  Use it to track when
```

Comment:
This type was intended to just be an ultra-lightweight non-opaque utility, removing the need to write `if wg != nil {` a bunch of places.

Note that since it is not opaque, code that needs other operations on the WaitGroup has them, without us needing to write any boilerplate here.

I would prefer to not promote this to a full first-class abstraction.

Reply:
Resolving this thread as the GoRoutineCounter seems to be the agreed on solution here.